lib/glific/wa_messages.ex

```diff
@@ -176,4 +179,48 @@ defmodule Glific.WAMessages do
       &[element | &1]
     )
   end
+
+  @spec add_empty_conversations([WAConversation.t()], map()) :: [WAConversation.t()]
+
+  defp add_empty_conversations(results, %{filter: %{id: id}}),
+    do: add_empty_conversation(results, [id])
+
+  defp add_empty_conversations(results, %{filter: %{ids: ids}}),
+    do: add_empty_conversation(results, ids)
+
+  defp add_empty_conversations(results, _), do: results
+
+  # helper function that actually implements the above functionality
+  @spec add_empty_conversation([WAConversation.t()], [integer]) :: [WAConversation.t()]
+  defp add_empty_conversation(results, wa_group_ids) when is_list(wa_group_ids) do
+    # first find all the group ids that we have some messages
+    present_wa_group_ids =
+      Enum.reduce(
+        results,
+        [],
+        fn r, acc -> [r.wa_group.id | acc] end
+      )
+
+    # the difference is the empty wa_group id list
+    empty_wa_group_ids = wa_group_ids -- present_wa_group_ids
+
+    # lets load all wa_group ids in one query, rather than multiple single queries
+    empty_results =
+      WAGroup
+      |> where([wa_grp], wa_grp.id in ^empty_wa_group_ids)
+      |> Repo.all()
+      # now only generate conversations objects for the empty wa_group ids
+      |> Enum.reduce(
+        [],
+        fn wa_group, acc -> add_conversation(acc, wa_group) end
+      )
+
+    results ++ empty_results
```

Comment:
why we are fetching wa_group_ids again. Ideally we should make one query to get a list of of top 25 wa_groups based on last_communication_at and then create conversations or empty conversations in one go

Reply:
We are not fetching the ids, we have the ids, we are fetching the wa_group struct. 

> Ideally we should make one query to get a list of of top 25 wa_groups based on last_communication_at and then create conversations or empty conversations in one go

We need to have wa_groups ordered wrto the latest messages. So sorting is not straightforward here.

What we could do is pass the top 25 wa_group struct list here and use as a cache.

What do you think?